code/datums/jobs/job/job.dm

```diff
@@ -218,13 +170,141 @@ GLOBAL_PROTECT(exp_specialmap)
+
+/datum/job/proc/manage_job_lists(previous_amount)
+	if(!previous_amount)
+		if(total_positions)
+			SSjob.add_active_occupation(src)
+	else if(!total_positions)
+		SSjob.remove_active_occupation(src)
+	if(!SSticker.HasRoundStarted() && previous_amount < total_positions)
+		LAZYADD(SSjob.occupations_reroll, src)
+
+
+// Spawning mobs.
+/mob/living/proc/apply_assigned_role_to_spawn(datum/job/assigned_role, client/player, datum/squad/assigned_squad)
+	job = assigned_role
+	skills = getSkillsType(job.return_skills_type(player?.prefs))
+	faction = job.faction
+	job.announce(src)
+
+
+/mob/living/carbon/human/apply_assigned_role_to_spawn(datum/job/assigned_role, client/player, datum/squad/assigned_squad)
+	. = ..()
+	comm_title = job.comm_title
+
+	if(job.outfit)
+		var/id_type = job.outfit.id ? job.outfit.id : /obj/item/card/id
+		var/obj/item/card/id/id_card = new id_type
+		if(wear_id)
+			stack_trace("[src] had an ID when apply_outfit_to_spawn() ran")
+			QDEL_NULL(wear_id)
+		equip_to_slot_or_del(id_card, SLOT_WEAR_ID)
+		job.outfit.handle_id(src)
+		job.outfit.equip(src)
+
+	if(job.job_flags & JOB_FLAG_ALLOWS_PREFS_GEAR)
+		equip_preference_gear(player)
+
+	if(!src.assigned_squad)
+		job.equip_spawning_squad(src, assigned_squad, player)
+
+
+/datum/job/proc/equip_spawning_squad(mob/living/carbon/human/new_character, datum/squad/assigned_squad, client/player)
 	return
+
+/datum/job/terragov/squad/equip_spawning_squad(mob/living/carbon/human/new_character, datum/squad/assigned_squad, client/player)
+	if(!assigned_squad)
+		SSjob.JobDebug("Failed to put marine role in squad. Player: [player.key] Job: [title]")
+		return
+	assigned_squad.insert_into_squad(new_character)
+
+
+/datum/job/proc/on_late_spawn(mob/living/late_spawner)
+	if(job_flags & JOB_FLAG_ADDTOMANIFEST)
+		if(!ishuman(late_spawner))
+			CRASH("on_late_spawn called for job with JOB_FLAG_ADDTOMANIFEST on non-human late_spawner: [late_spawner]")
+		GLOB.datacore.manifest_inject(late_spawner)
+	if(job_flags & JOB_FLAG_PROVIDES_BANK_ACCOUNT)
+		var/datum/money_account/bank_account = create_account(late_spawner.real_name, rand(50, 500) * 10)
+		var/list/remembered_info = list()
+		remembered_info += "<b>Your account number is:</b> #[bank_account.account_number]"
+		remembered_info += "<b>Your account pin is:</b> [bank_account.remote_access_pin]"
+		remembered_info += "<b>Your account funds are:</b> $[bank_account.money]"
+		if(length(bank_account.transaction_log))
+			var/datum/transaction/transaction_datum = bank_account.transaction_log[1]
+			remembered_info += "<b>Your account was created:</b> [transaction_datum.time], [transaction_datum.date] at [transaction_datum.source_terminal]"
+		late_spawner.mind.store_memory(remembered_info.Join("<br>"))
+		late_spawner.mind.initial_account = bank_account
+
+/datum/job/proc/return_spawn_type(datum/preferences/prefs)
+	return /mob/living/carbon/human
+
+/datum/job/proc/return_skills_type(datum/preferences/prefs)
```

Comment:
why does this take prefs?

Reply:
Synths may have an early synth pref. The job is the same, but the pref alters the skills.

Reply:
Ohhh TIL